test/extended/builds/volumes.go

```diff
@@ -0,0 +1,203 @@
+			expectedBuildVolumes := []buildv1.BuildVolume{
+				{
+					Name: "this-secret",
+					Source: buildv1.BuildVolumeSource{
+						Type: buildv1.BuildVolumeSourceTypeSecret,
+						Secret: &corev1.SecretVolumeSource{
+							SecretName:  "my-secret",
+							DefaultMode: &defaultMode,
+						},
+					},
+					Mounts: []buildv1.BuildVolumeMount{
+						{
+							DestinationPath: "/var/run/secrets/some-secret",
+						},
+					},
+				},
+				{
+					Name: "this-configmap",
+					Source: buildv1.BuildVolumeSource{
+						Type: buildv1.BuildVolumeSourceTypeConfigMap,
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: "my-configmap"},
+							DefaultMode:          &defaultMode,
+						},
+					},
+					Mounts: []buildv1.BuildVolumeMount{
+						{
+							DestinationPath: "/var/run/configmaps/some-configmap",
+						},
+					},
+				},
+			}
+			expectedVolumes := []corev1.Volume{
+				{
+					Name: "my-secret-user-build-volume",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName:  "my-secret",
+							DefaultMode: &defaultMode,
+						},
+					},
+				},
+				{
+					Name: "my-configmap-user-build-volume",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: "my-configmap"},
+							DefaultMode:          &defaultMode,
+						},
+					},
+				},
+			}
+			expectedVolumeMounts := []corev1.VolumeMount{
+				{
+					Name:      "my-secret-user-build-volume",
+					MountPath: "/var/run/openshift.io/volumes/my-secret-user-build-volume",
+					ReadOnly:  true,
+				},
+				{
+					Name:      "my-configmap-user-build-volume",
+					MountPath: "/var/run/openshift.io/volumes/my-configmap-user-build-volume",
+					ReadOnly:  true,
+				},
+			}
+
+			for _, strategyType := range []string{"s2i", "docker"} {
+				buildConfigName := fmt.Sprintf("my%stest", strategyType)
+
+				g.By(fmt.Sprintf("creating build config for %q", buildConfigName))
+				err := oc.Run("create").Args("-f", filepath.Join(baseDir, fmt.Sprintf("buildconfig-%s.yaml", strategyType))).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				bc, err := oc.BuildClient().BuildV1().BuildConfigs(oc.Namespace()).Get(context.Background(), buildConfigName, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By(fmt.Sprintf("updating build config %q script and build volumes", buildConfigName))
+				bc.Spec.PostCommit.Script = script
+				switch bc.Spec.Strategy.Type {
+				case buildv1.SourceBuildStrategyType:
+					bc.Spec.Strategy.SourceStrategy.Volumes = expectedBuildVolumes
+				case buildv1.DockerBuildStrategyType:
+					bc.Spec.Strategy.DockerStrategy.Volumes = expectedBuildVolumes
+				}
+				bc, err = oc.BuildClient().BuildV1().BuildConfigs(oc.Namespace()).Update(context.Background(), bc, metav1.UpdateOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
e2e test should just have one create/apply call - easier to do with separate `g.It` clauses

Reply:
So .... Do you want me to incorporate the build volumes stuff into the yaml so that it is part of the build config on create?
I would love to set a precedent of having base build configs and such as yaml files that we use, and then within the test patch or update them with whatever is needed for the test. That also makes it easier to keep in sync.

Reply:
As unwieldy as YAML is, I prefer our e2e's to test our code the same way we ask customers to use our code. `oc apply -f <YAML>` is the preferred approach for most APIs, with notable exceptions for some of our helper commands (new-build, start-build, etc.)

Reply:
updated in current push